### PR TITLE
Improve the getVersions flow in odsp driver

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

We never use the trees/latest cache/network fetch flow in getVersions in case it is not first call. Which leads to two calls first to getVersions and then to fetch snapshots. With this change we will still utilize the either cache or just 1 network call in case a specific version is not designed.

### CODE REVIEW

File: packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts [hunk 1]
```diff
@@ -275,7 +274,10 @@ export class OdspDocumentStorageService extends OdspDocumentStorageServiceBase {
                         // Based on the concurrentSnapshotFetch policy:
                         // Either retrieve both the network and cache snapshots concurrently and pick the first to return,
                         // or grab the cache value and then the network value if the cache value returns undefined.
-                        if (this.hostPolicy.concurrentSnapshotFetch && !this.hostPolicy.summarizerClient) {
+                        // For summarizer which could call this a lot due to refreshing of summary parent, always use the cache
```
Comment (NicholasCouri) [inline packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:277]: >could call this a lot due to refreshing of summary parent [](http://example.com/codeflow?start=48&length=57)

is it true ? I would expect it not to be called a lot or ever as long as we use the pendngsummaries as we would skip the refresh....

Comment (jatgarg) [reply packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts:277]: I will change the wording. A lot may not be the right word.

**Review by NicholasCouri — APPROVED** [review]
:shipit:

**Comment by vladsud** [comment]
@jatgarg , are you sure you want to populate blob cache for the second call? What is the scenario where it happens? If it's mostly summarizer, then I think we do not use blobs in such case, so keeping them around would be a waste of memory, no?

**Comment by jatgarg** [comment]
@vladsud I think I missed that. I will correct that and not fetch/cache blobs if it is not the first call.

**Pull request merged**